src/main/java/me/moodcat/database/BulkInsertData.java

```diff
@@ -0,0 +1,219 @@
+import me.moodcat.database.controllers.ArtistDAO;
+import me.moodcat.database.controllers.RoomDAO;
+import me.moodcat.database.controllers.SongDAO;
+import me.moodcat.database.embeddables.VAVector;
+import me.moodcat.database.entities.Artist;
+import me.moodcat.database.entities.ChatMessage;
+import me.moodcat.database.entities.Room;
+import me.moodcat.database.entities.Song;
+import me.moodcat.soundcloud.SoundCloudException;
+import me.moodcat.soundcloud.SoundCloudExtract;
+import me.moodcat.soundcloud.SoundCloudTrack;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+import com.google.inject.persist.Transactional;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.sql.ordering.antlr.Factory;
+
+/**
+ * MockData inserts initial data in to a clean database.
+ *
+ * @author Jan-Willem Gmelig Meyling
+ */
+@Singleton
+public class BulkInsertData {
+
+    /**
+     * The default [0,0] VA vector.
+     */
+    private static final VAVector DEFAULT_VECTOR = new VAVector(0, 0);
+
+    /**
+     * The path to a newline seperated list of SoundCloud ids.
+     */
+    private static final String SOUNDCLOUD_ID_FILE_PATH = "./src/main/resources/soundcloud_ids";
+
+    /**
+     * The artist DAO provider.
+     */
+    private final Provider<ArtistDAO> artistDAOProvider;
+
+    /**
+     * The song DAO provider.
+     */
+    private final Provider<SongDAO> songDAOProvider;
+
+    /**
+     * The room DAO provider.
+     */
+    private final Provider<RoomDAO> roomDAOProvider;
+
+    /**
+     * The SoundCloud extraction API.
+     */
+    private final SoundCloudExtract soundCloudExtract;
+
+    /**
+     * Object to bulk insert a list of given songs into the databse.
+     * 
+     * @param artistDAOProvider
+     *            the artist provider.
+     * @param songDAOProvider
+     *            the song provider.
+     * @param roomDAOProvider
+     *            the room provider.
+     */
+    @Inject
+    public BulkInsertData(final Provider<ArtistDAO> artistDAOProvider,
+            final Provider<SongDAO> songDAOProvider,
+            final Provider<RoomDAO> roomDAOProvider) {
+        this.artistDAOProvider = artistDAOProvider;
+        this.songDAOProvider = songDAOProvider;
+        this.roomDAOProvider = roomDAOProvider;
+        soundCloudExtract = new SoundCloudExtract();
+    }
+
+    /**
+     * Retrieve songs from the API and put the in the database.
+     * 
+     * @throws IOException
+     *             when the file with song ids could not be parsed.
+     * @throws SoundCloudException
+     *             when the SoundCloud API was not called successfully.
+     */
+    @Transactional
+    public void insertData() throws IOException, SoundCloudException {
+        final SongDAO songDAO = songDAOProvider.get();
+        HashMap<String, Artist> artistMap = new HashMap<>();
```

Comment:
You need a `Map` and do not necessarily require hash map behaviour. So it might be an idea to change the type of the parameters/variables to `Map` (and still instantiate it with hashmaps).
Not really a required change in order for merge, just a thought.